lib/libYulAST/YulContractNode.cpp

```diff
@@ -114,28 +133,76 @@ YulContractNode::getFunctions() {
 
 std::string YulContractNode::to_string() { return "contract"; }
 
-std::map<std::string, StorageVarInfo> &YulContractNode::getVarTypeMap() {
-  return varTypeMap;
-}
-
 std::map<std::string, TypeInfo> &YulContractNode::getTypeInfoMap() {
   return typeInfoMap;
 }
 
-std::string YulContractNode::getStateVarNameBySlotOffset(int slot, int offset) {
-  for (auto &f : structFieldOrder) {
-    auto varEntry = varTypeMap[f];
-    int varSlot = varEntry.slot;
-    int varOffset = varEntry.offset;
-    if (varOffset == offset && varSlot == slot) {
-      return f;
+unsigned int YulContractNode::getFieldIndexInStruct(TypeInfo ti,
+                                                    std::string name) {
+  assert(ti.kind == "struct" && "type not struct");
+  unsigned int i = 0;
+  for (auto mem : ti.members) {
+    if (name == mem.name) {
+      return i;
+    }
+    i++;
+  }
+  assert(false && "field member not found");
+  return INT_MAX;
```

Comment:
Would be better to use `std::numeric_limits` from `<limits>`
```suggestion
  return std::numeric_limits<int>::max();
```

Reply:
done